wgengine/magicsock/magicsock.go

```diff
@@ -2233,6 +2233,16 @@ func (c *Conn) bindSocket(ruc *RebindingUDPConn, network string, curPortFate cur
 			continue
 		}
 		trySetSocketBuffer(pconn, c.logf)
+
+		if CanPMTUD() {
+			err = setDontFragment(pconn, network)
+			if err != nil {
+				c.logf("magicsock: set dontfragment failed for %v port %d: %v", network, port, err)
+				// TODO disable PMTUD in this case. We don't expect the setsockopt to fail on
```

Comment:
What happens on Android? Have they allowlisted the sockopt?

Reply:
That's a good question. I think the unfortunately even more subtle answer is that we want to disable PMTU when we're on battery and on some LTE or lesser protocol radios. Those are definitely harder but also quite interesting conditions to determine so I suspect we'll want to defer them. Upcoming pervasive 5G deployments, and wifi could benefit, and so it is worth considering in time.

Reply:
FWIW, this is where I would introduce a separation between "can" and "should", and "can" would be set here on failure to to "can not"

Reply:
That makes sense. I think we should change the build to exclude android for now and then get fancy with battery, LTE, wifi later on.